src/Sylius/Bundle/UserBundle/Controller/UserController.php

```diff
@@ -105,7 +110,91 @@ public function resetPasswordAction(Request $request, $token)
         );
     }
 
-    protected function prepareResetPasswordRequest(Request $request, TokenProviderInterface $generator, $senderEvent)
+    /**
+     * @param Request $request
+     * @param string $token
+     *
+     * @return Response
+     */
+    public function verifyAction(Request $request, $token)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+        $redirectRoute = $request->attributes->get('_sylius[redirect]', 'sylius_shop_homepage', true);
+        $response = $this->redirectToRoute($redirectRoute);
+
+        /** @var UserInterface $user */
+        $user = $this->repository->findOneBy(['emailVerificationToken' => $token]);
+        if (null === $user) {
+            if (!$configuration->isHtmlRequest()) {
+                return $this->viewHandler->handle($configuration, View::create($configuration, Response::HTTP_BAD_REQUEST));
+            }
+
+            $this->addFlash('error', 'sylius.user.verification.error');
+
+            return $this->redirectToRoute('sylius_shop_homepage');
+        }
+
+        $eventDispatcher = $this->container->get('event_dispatcher');
```

Comment:
You can use the resource controller's event dispatcher: ``$this->eventDispatcher``.